internal/sleuth/incident_impact_source_resource.go

```diff
@@ -0,0 +1,876 @@
+	diags.Append(asDiags...)
+
+	var jiraInput *jiraInputResourceModel
+	asDiags = plan.JiraInput.As(ctx, &jiraInput, basetypes.ObjectAsOptions{})
+	diags.Append(asDiags...)
+
+	var blamelessInput *blamelessInputResourceModel
+	asDiags = plan.BlamelessInput.As(ctx, &blamelessInput, basetypes.ObjectAsOptions{})
+	diags.Append(asDiags...)
+
+	var statuspageInput *statuspageInputResourceModel
+	asDiags = plan.StatusPageInput.As(ctx, &statuspageInput, basetypes.ObjectAsOptions{})
+	diags.Append(asDiags...)
+
+	var opsgenieInput *opsgenieInputResourceModel
+	asDiags = plan.OpsGenieInput.As(ctx, &opsgenieInput, basetypes.ObjectAsOptions{})
+	diags.Append(asDiags...)
+
+	var firehydrantInput *firehydrantInputResourceModel
+	asDiags = plan.FireHydrantInput.As(ctx, &firehydrantInput, basetypes.ObjectAsOptions{})
+	diags.Append(asDiags...)
+
+	var clubhouseInput *clubhouseInputResourceModel
+	asDiags = plan.ClubhouseInput.As(ctx, &clubhouseInput, basetypes.ObjectAsOptions{})
+	diags.Append(asDiags...)
+
+	return providerData{
+		pagerduty:   pdInput,
+		datadog:     ddInput,
+		jira:        jiraInput,
+		blameless:   blamelessInput,
+		statuspage:  statuspageInput,
+		opsgenie:    opsgenieInput,
+		firehydrant: firehydrantInput,
+		clubhouse:   clubhouseInput,
+	}, diags
+
+}
+
+func (iisr *incidentImpactSourceResource) Create(ctx context.Context, req resource.CreateRequest, res *resource.CreateResponse) {
+	ctx = tflog.SetField(ctx, "resource", "incident_impact_source")
+	ctx = tflog.SetField(ctx, "operation", "create")
+
+	var plan incidentImpactResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	res.Diagnostics.Append(diags...)
+
+	tflog.Info(ctx, "Creating IncidentImpactSource resource", map[string]any{"name": plan.Name.ValueString(), "projectSlug": plan.ProjectSlug.ValueString()})
+	var pdInput *pagerDutyInputResourceModel
+
+	tflog.Error(ctx, fmt.Sprintf("REMOVE ME! %+v", plan.PagerDutyInput.IsNull()))
+
+	diags = plan.PagerDutyInput.As(ctx, &pdInput, basetypes.ObjectAsOptions{})
+	res.Diagnostics.Append(diags...)
+
+	pd, diags := parseProviderData(ctx, plan)
+
+	res.Diagnostics.Append(diags...)
+
+	if res.Diagnostics.HasError() {
+		tflog.Error(ctx, "Error getting IncidentImpactSource plan", map[string]any{"diagnostics": res.Diagnostics})
+		return
+	}
+
+	projectSlug := plan.ProjectSlug.ValueString()
+	input, diags := getMutableIncidentImpactSourceStruct(ctx, plan, pd)
+	res.Diagnostics.Append(diags...)
+	//input := gqlclient.IncidentImpactSourceInputType{
+	//	ProjectSlug:     projectSlug,
+	//	Name:            plan.Name.ValueString(),
+	//	Provider:        strings.ToUpper(plan.ProviderName.ValueString()),
+	//	EnvironmentName: strings.ToLower(plan.EnvironmentName.ValueString()),
+	//}
+	//if pdInput != nil {
+	//	input.PagerDutyInputType = &gqlclient.PagerDutyInputType{
+	//		RemoteServices: pdInput.RemoteServices.ValueString(),
+	//		RemoteUrgency:  pdInput.RemoteUrgency.ValueString(),
+	//	}
+	//}
```

Comment:
What is the reason for this being in here, but commented out?

Reply:
ah good catch, will remove

Reply:
this was just a first MVP before extracting logic for others out 👍